nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2122,17 +2121,13 @@ public void setLegalHold(DocumentRef docRef, boolean hold, String comment) {
         DocumentModel docModel = readModel(doc);
         Map<String, Serializable> options = new HashMap<>();
         options.put(CoreEventConstants.LEGAL_HOLD, Boolean.valueOf(hold));
-        String commentStart = Boolean.toString(hold);
-        if (comment == null) {
-            comment = commentStart;
-        } else {
-            comment = commentStart + " " + comment;
-        }
-        options.put("comment", comment);
-        notifyEvent(DocumentEventTypes.BEFORE_SET_LEGAL_HOLD, docModel, options, null, null, true, false);
+        options.put("comment", comment != null ? comment : "");
```

Comment:
In the platform we prefer to not have negations in ternary operators, so `comment == null ? "" : comment`. But actually in this case it would be even better to use the semantically cleaner `StringUtils.defaultString(comment)`